crates/iota-framework/packages/iota-system/tests/validator_set_tests.move

```diff
@@ -79,62 +96,307 @@ module iota_system::validator_set_tests {
         let ctx = scenario.ctx();
-        let v1 = create_validator(@0x1, 1, 45, true, ctx);
-        let v2 = create_validator(@0x2, 2, 42, false, ctx);
-        let v3 = create_validator(@0x3, 3, 40, false, ctx);
-        let v4 = create_validator(@0x4, 4, 41, false, ctx);
-        let v5 = create_validator(@0x5, 10, 43, false, ctx);
-        // Create a validator set with only the first validator in it.
-        let mut validator_set = validator_set::new(vector[v1], ctx);
-
-        assert_eq(validator_set.derive_reference_gas_price(), 45);
-
-        add_and_activate_validator(&mut validator_set, v2, scenario);
-        advance_epoch_with_dummy_rewards(&mut validator_set, scenario);
-
-        assert_eq(validator_set.derive_reference_gas_price(), 45);
-
-        add_and_activate_validator(
-            &mut validator_set,
-            v3,
-            scenario
-        );
-        advance_epoch_with_dummy_rewards(&mut validator_set, scenario);
-
-        assert_eq(validator_set.derive_reference_gas_price(), 42);
+        let v1 = create_validator_with_stake(@0x1, 2, 1, 28 * NANOS_PER_IOTA, true, ctx);
+        let v2 = create_validator_with_stake(@0x2, 4, 1, 4 * NANOS_PER_IOTA, true, ctx);
+        let v3 = create_validator_with_stake(@0x3, 6, 1, 22 * NANOS_PER_IOTA, true, ctx);
+        let v4 = create_validator_with_stake(@0x4, 8, 1, 8 * NANOS_PER_IOTA, true, ctx);
+        let v5 = create_validator_with_stake(@0x5, 20, 1, 24 * NANOS_PER_IOTA, false, ctx);
+        let v6 = create_validator_with_stake(@0x6, 22, 1, 22 * NANOS_PER_IOTA, false, ctx);
+        let v7 = create_validator_with_stake(@0x7, 24, 1, 24 * NANOS_PER_IOTA, false, ctx);
+        let v8 = create_validator_with_stake(@0x8, 3, 2, 3 * NANOS_PER_IOTA, false, ctx);
+        let v9 = create_validator_with_stake(@0x9, 28, 1, 28 * NANOS_PER_IOTA, false, ctx);
+
+        let committee_size = 5;
+
+        // Create a validator set with all validators in it, to check that regardless of the active_validators order, top stakers are selected correctly.
+        let validator_set_instance = validator_set::new_v2(vector[v1, v2, v3, v4, v5, v6, v7, v8, v9], committee_size, ctx);
+  
+        assert_same_elems(validator_set_instance.active_validator_addresses(), vector[@0x1, @0x2, @0x3, @0x4, @0x5, @0x6, @0x7, @0x8, @0x9]);
+        assert_eq(validator_set_instance.committee_validator_addresses(),  vector[@0x9, @0x1, @0x7, @0x5, @0x6]);
+        assert_eq(validator_set_instance.total_stake_inner(), (28 + 28 + 24 + 24 + 22) * NANOS_PER_IOTA);
+
+        test_utils::destroy(validator_set_instance);
+        scenario_val.end();
+    }
 
-        add_and_activate_validator(
-            &mut validator_set,
-            v4,
-            scenario
-        );
-        advance_epoch_with_dummy_rewards(&mut validator_set, scenario);
 
-        assert_eq(validator_set.derive_reference_gas_price(), 42);
+    // TODO: implement tests that check max_committee_members_count changes
```

Comment:
What about this?

Reply:
Implemented the test, forgot to remove the todo.